lib/src/adapters/persistent_json_api.dart

```diff
@@ -193,42 +266,75 @@ class PersistentJsonApiAdapter extends JsonApiAdapter {
     });
   }
 
-  Future<int> storeAdd(String endpoint, JsonApiDocument doc) async {
-    var store = openIntKeyStore('added');
-    int id = await store.add(database, _storeAdapter.toMap(doc));
-    return id;
+  Future<String> storeAdd(String endpoint, JsonApiDocument doc) async {
+    final tempId = doc.id ?? Uuid().v1();
+    doc.id = tempId;
+    storePutOne(endpoint, doc);
+
+    var store = openAddedStore();
+    await store.record(tempId).delete(database);
+    await store.record(tempId).add(database, {'endpoint': endpoint});
+    return tempId;
+  }
+
+  Future<void> storeUpdate(String endpoint, JsonApiDocument doc) async {
+    storePutOne(endpoint, doc);
+
+    var store = openUpdatedStore();
+    await store.record(doc.id!).delete(database);
+    await store.record(doc.id!).add(database, {'endpoint': endpoint});
+  }
+
+  Future<void> storeDelete(String endpoint, JsonApiDocument doc) async {
+    storePutOne(endpoint, doc);
+
+    var store = openRemovedStore();
+    await store.record(doc.id!).delete(database);
+    await store.record(doc.id!).add(database, {'endpoint': endpoint});
   }
 
   Future<Iterable<JsonApiDocument>> addedByEndpoint(String endpoint) async {
-    var store = openIntKeyStore('added');
-    var docs = await store.find(
+    var addedStore = openAddedStore();
+
+    var addedIds = (await addedStore.findKeys(
       database,
-      finder: Finder(filter: Filter.matches('endpoint', endpoint)),
-    );
-    return docs.map((e) => _storeAdapter.fromMap(e.value)).toList();
+      finder: Finder(filter: Filter.equals('endpoint', endpoint)),
+    ));
+
+    var store = openStringKeyStore(endpoint);
+    var list = (await store.records(addedIds).get(database))
+        .where((e) => e != null)
+        .map((e) => _storeAdapter.fromMap(e!))
+        .toList();
+
+    return list;
   }
 
   Future<JsonApiDocument?> findAdded(String id) async {
-    var key = int.parse(id.replaceAll('added:', ''));
-    var store = openIntKeyStore('added');
-    var map = await store.record(key).get(database);
-    return map == null ? null : _storeAdapter.fromMap(map);
+    var store = openAddedStore();
+    var added = await store.record(id).get(database);
+    if (added == null) return null;
+    return peek(added['endpoint'].toString(), id);
```

Comment:
@Bringoff Why are we using `peek()` here? Shouldn't added documents be stored and retrieved on the local db?

Reply:
@algodave right, thanks for pointing that out, I actually intended to use `storeGetOne` here. Fixed